src/Action.hs

```diff
@@ -104,12 +110,6 @@ instance PersistM r => ActionPersist r (Action r) where
 instance MonadIO (Action r) where
     liftIO = Action . liftIO
 
-instance PersistM r => PersistM (Action r) where
```

Comment:
You are right. Probably changing the `Monad` in `Authorize` will do the trick.

````haskell
class Authorize a where
    authorizeGet :: PersistM m => Username -> AulaGetter a -> m ()
    authorizeSet :: PersistM m => Username -> AulaSetter a -> m ()
````